From b0bd8ffecd70b46e584e2e52ea74f2373b544217 Mon Sep 17 00:00:00 2001
From: LOLi <loli10K@users.noreply.github.com>
Date: Thu, 3 Aug 2017 18:43:17 +0200
Subject: [PATCH] Fix parsable 'zfs get' for compressratios
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

This is consistent with the change introduced in bc2d809 where
'zpool get -p dedupratio' does not add a trailing "x" to the output.

Reviewed-by: Tony Hutter <hutter2@llnl.gov>
Reviewed-by: Giuseppe Di Natale <dinatale2@llnl.gov>
Signed-off-by: loli10K <ezomori.nozomu@gmail.com>
Closes #6436
Closes #6449
---
 lib/libzfs/libzfs_dataset.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

Index: zfs-linux-0.7.9/lib/libzfs/libzfs_dataset.c
===================================================================
--- zfs-linux-0.7.9.orig/lib/libzfs/libzfs_dataset.c
+++ zfs-linux-0.7.9/lib/libzfs/libzfs_dataset.c
@@ -2611,9 +2611,14 @@ zfs_prop_get(zfs_handle_t *zhp, zfs_prop
 	case ZFS_PROP_COMPRESSRATIO:
 		if (get_numeric_property(zhp, prop, src, &source, &val) != 0)
 			return (-1);
-		(void) snprintf(propbuf, proplen, "%llu.%02llux",
-		    (u_longlong_t)(val / 100),
-		    (u_longlong_t)(val % 100));
+		if (literal)
+			(void) snprintf(propbuf, proplen, "%llu.%02llu",
+			    (u_longlong_t)(val / 100),
+			    (u_longlong_t)(val % 100));
+		else
+			(void) snprintf(propbuf, proplen, "%llu.%02llux",
+			    (u_longlong_t)(val / 100),
+			    (u_longlong_t)(val % 100));
 		break;
 
 	case ZFS_PROP_TYPE:
